src/server.rs

```diff
@@ -99,15 +109,25 @@ async fn handle_spec_request(
     return Err(Error::GetSpecFail);
 }
 
-fn all_routes(state: Arc<(Aws, Vec<String>, String)>) -> Router {
+fn all_routes<'a>(
+    state: Arc<(
+        impl InfraProvider + Send + Sync + Clone + 'static,
+        Vec<String>,
+        String,
+    )>,
+) -> Router {
     Router::new()
         .route("/ip", get(handle_ip_request))
         .route("/spec", get(handle_spec_request))
         .with_state(state)
 }
 
-pub async fn serve(client: Aws, regions: Vec<String>, rates_path: String) {
-    let state: Arc<(Aws, Vec<String>, String)> = Arc::from((client, regions, rates_path));
+pub async fn serve(
+    client: impl InfraProvider + Send + Sync + Clone + 'static,
+    regions: Vec<String>,
+    rates_path: String,
```

Comment:
use the rates read in main instead of reading again to ensure consistency between server responses and what the control plane uses

Reply:
Done